packages/contracts-bedrock/src/Safe/VetoerSet/OwnerGuard.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { FixedPointMathLib } from "solady/utils/FixedPointMathLib.sol";
+
+import { Safe } from "safe-contracts/Safe.sol";
+import { BaseGuard } from "safe-contracts/base/GuardManager.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+
+import { ISemver } from "src/universal/ISemver.sol";
+
+/// @title OwnerGuard
+/// @notice This Guard contract is used to enforce a maximum number of owners and a required
+///         threshold for the Safe Account.
+contract OwnerGuard is ISemver, BaseGuard {
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @notice The Safe Account for which this contract will be the guard.
+    Safe public immutable safe;
+
+    /// @notice The maximum number of owners. Can be changed by supermajority.
+    uint8 public maxOwnerCount;
+
+    /// @notice Thrown at deployment if the current Safe Account owner count can't fit in a `uint8`.
+    /// @param ownerCount The current owner count.
+    error OwnerCountTooHigh(uint256 ownerCount);
+
+    /// @notice Thrown if the new owner count is above the `maxOwnerCount` limit.
+    /// @param ownerCount The Safe Account owner count.
+    /// @param maxOwerCount The current `maxOwnerCount`.
+    error InvalidOwnerCount(uint256 ownerCount, uint256 maxOwerCount);
+
+    /// @notice Thrown after the Safe Account executed a transaction if its threshold does not matches
+    ///         with the desired 66% threshold.
+    /// @param threshold The Safe Account threshold.
+    /// @param expectedThreshold The expected threshold.
+    error InvalidSafeAccountThreshold(uint256 threshold, uint256 expectedThreshold);
+
+    /// @notice Thrown when trying to update the `maxOwnerCount` limit but the caller is not
+    ///         the associated Safe Account.
+    /// @param sender The sender address.
+    error SenderIsNotSafeAccount(address sender);
+
+    /// @notice Thrown when trying to update the `maxOwnerCount` limit to a value lower than the current
+    ///         Safe Account owner count.
+    /// @param newMaxOwnerCount The new invalid max count.
+    /// @param ownerCount The current Safe Account owner count.
+    error InvalidNewMaxCount(uint256 newMaxOwnerCount, uint256 ownerCount);
```

Comment:
Following up from the earlier comment, prefer to be more explicit about the condition making the count invalid here
```suggestion
    error MaxCountTooLow(uint256 newMaxOwnerCount, uint256 ownerCount);
```

Reply:
Done, but instead of `MaxCountTooLow` I went with `MaxOwnerCountTooLow`.